src/amuse/support/argparse.py

```diff
@@ -0,0 +1,68 @@
+import argparse
```

Comment:
It's not clear to me how this replaces the code in `units.optparse`. That code seems to be a helper for doing additional type checking of arbitrary arguments, and for improving formatting of the help text, while this defines (and standardises?) a number of specific arguments that can be passed to AMUSE scripts. So is this a replacement or an addition?

Reply:
#172 (I think) details how optparse is obsolete- attaching a unit is now possible by specifying type=units.(unit). This PR explicitly makes optparse deprecated, encouraging the use of argparse and making our literature arguments more accessible. Also, the new_argument_parser here will show the default settings, which was missing before.